src/query/service/src/pipelines/processors/transforms/new_hash_join/memory/basic.rs

```diff
@@ -135,20 +135,19 @@ impl BasicHashJoin {
             bytes: num_bytes,
         }))
     }
-}
-
-impl BasicHashJoin {
-    fn steal_chunk_index(&self) -> Option<usize> {
-        let locked = self.state.mutex.lock();
-        let _locked = locked.unwrap_or_else(PoisonError::into_inner);
-        self.state.build_queue.as_mut().pop_front()
-    }
 
     pub(crate) fn finalize_chunks(&mut self) {
         if self.desc.build_projection.is_empty() || !self.state.columns.is_empty() {
             return;
         }
 
+        if matches!(
```

Comment:
Perhaps we can move the branch logic into the nested loop join implementation? I see it only prevents the real finalize build logic. Please let me know if I'm missing any critical information.